hivemq-edge/src/frontend/src/modules/ProtocolAdapters/components/drawers/AdapterInstanceDrawer.tsx

```diff
@@ -78,6 +87,22 @@ const AdapterInstanceDrawer: FC<AdapterInstanceDrawerProps> = ({
     return errors.filter((error) => !error.stack.startsWith('no schema with key or ref'))
   }
 
+  const context: AdapterContext = {
+    onBatchUpload: (idSchema: IdSchema<unknown>, batch) => {
+      const path = idSchema.$id.replace('root_', '/').replaceAll('_', '/') + '/-'
+      const operations: JSONPatchDocument = batch.map<JSONPatchAdd>((value) => ({ op: 'add', path, value }))
+
+      if (schema && FLAG_POST_VALIDATE) {
+        const updatedDocument = immutableJSONPatch(defaultValues, operations)
+        const { $schema, ...rest } = schema
+        const validate = validator.ajv.compile(rest)
+        validate(updatedDocument)
+      }
+
+      setBachData(operations)
```

Comment:
suggestion: found a small typo

```suggestion
      setBatchData(operations)
```